File: array/plus_one/plus_one.rb
def plus_one(digits)
  n = digits.length
  (0...n).reverse_each do |i|
    if digits[i] < 9
      digits[i] += 1
      return digits
    end
    digits[i] = 0
  end
  new_num = Array.new(n + 1) { 0 }
  new_num[0] = 1
  new_num
end

a = [8, 8, 9]
p('input', a)
p('output', plus_one(a))

b = [9, 9, 9, 9]
p('input', b)
p('output', plus_one(b))

File: array/plus_one/plus_one.py
from typing import List


def plus_one(digits: List[int]):
    n = len(digits)
    for i in range(n - 1, -1, -1):
        if digits[i] < 9:
            digits[i] += 1
            return digits
        digits[i] = 0
    new_num = [0] * (n + 1)
    new_num[0] = 1
    return new_num


a = [8, 8, 9]
print('input', a)
print('output', plus_one(a))

b = [9, 9, 9, 9]
print('input', b)
print('output', plus_one(b))
